src/App.tsx

```diff
@@ -8,53 +9,102 @@
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsIsLoading, setPostsIsLoading] = useState(false);
+  const [hasPostsError, setHasPostsError] = useState('');
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [chosenPost, setChosenPost] = useState<Post | null>(null);
 
-              <Loader />
+  const hasNoPosts =
+    selectedUser && !postsIsLoading && !hasPostsError && !posts.length;
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  useEffect(() => {
+    if (!selectedUser) {
+      return;
+    }
+
+    setPostsIsLoading(true);
+    setHasPostsError('');
+    setPosts([]);
+    setChosenPost(null);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    getUserPosts(selectedUser.id)
+      .then(setPosts)
+      .catch(() => setHasPostsError('Something went wrong!'))
+      .finally(() => setPostsIsLoading(false));
+  }, [selectedUser]);
+
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                <UserSelector
+                  selectedUser={selectedUser}
+                  onSelect={setSelectedUser}
+                  onPost={setChosenPost}
+                />
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                {postsIsLoading && <Loader />}
+
+                {!postsIsLoading && hasPostsError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {hasPostsError}
+                  </div>
+                )}
+
+                {hasNoPosts && (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                )}
+
+                {!postsIsLoading && posts.length > 0 && (
+                  <PostsList
+                    posts={posts}
+                    choosenPost={chosenPost}
```

Comment:
There is a typo in the prop name `choosenPost`. It should be `chosenPost` to match the state variable name and maintain consistency.